docs/en/04_Changelogs/4.0.0.md

```diff
@@ -28,12 +85,187 @@ Note that this will not allow you to utilise certain file versioning features in
 
 	:::yaml
 	\SilverStripe\Filesystem\Flysystem\FlysystemAssetStore:
-		legacy_paths: true
+	  legacy_paths: true
 
 
 See [/developer_guides/files/file_management] for more information on how the new system works.
 
 
+### Upgrade code which acts on `Image`
+
+As all image-specific manipulations has been refactored from `Image` into an `ImageManipulations` trait, which 
+is applied to both `File` and `DBFile`. These both implement a common interface `AssetContainer`, which
+has the `getIsImage()` method. In some cases, it may be preferable to invoke this method to detect
+if the asset is an image or not, rather than checking the subclass, as the asset may also be a `DBFile` with
+an image filter applied, rather than an instance of the `Image` dataobject.
+
+In addition, a new file category `image/supported` has been added, which is a subset of the `image` category.
+This is the subset of all image types which may be assigned to the `[api:Image]` dataobject, and may have
+manipulations applied to it. This should be used as the file type restriction on any `[api:UploadField]` which
+is intended to upload images for manipulation. All image categories have also been specified via constants
+on the `[api:File]` DataObject.
```

Comment:
What's the rationale on using constants here? It makes configuration inconsistent, since you still need to use the string keys ("image" rather than `File::CATEGORY_IMAGE`) for any YAML config via `File.app_categories`. We could just check that a string key exists in `File.app_categories`? Moves it from a static to a runtime check, but more consistent between PHP and YAML use. It also means devs can introduce new categories easily - they couldn't create a new constant in `File`.